src/middlewared/middlewared/plugins/iscsi_/host_injection.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+from collections import namedtuple
+from ipaddress import ip_address
+import os
+
+from middlewared.service import Service
+
+CollectedHost = namedtuple("CollectedHost", ["ip", "iqn"])
+
+
+class iSCSIHostsInjectionService(Service):
+    class Config:
+        namespace = "iscsi.host.injection"
+        private = True
+
+    async def run(self):
+        while True:
+            try:
+                await self.middleware.call(
+                    "iscsi.host.batch_update",
+                    [
+                        host._asdict()
+                        for host in await self.middleware.call("iscsi.host.injection.collect")
+                    ],
```

Comment:
For these I think we would like an indication wether the host was added automatically or manually?